source/fundamentals/enterprise-auth.txt

```diff
@@ -0,0 +1,216 @@
+information on establishing a connection to your MongoDB cluster, read our
+:doc:`Connection Guide </fundamentals/connection>`.
+
+
+Specify an Authentication Mechanism
+-----------------------------------
+
+.. include:: /includes/fundamentals/auth-specify.rst
+
+Mechanisms
+----------
+
+.. _gssapi-auth-mechanism:
+
+Kerberos (GSSAPI/SSPI)
+~~~~~~~~~~~~~~~~~~~~~~
+
+*Available in MongoDB Enterprise Edition 3.7 and later.*
+
+The Generic Security Services API (``GSSAPI``) authentication mechanism
+allows the user to authenticate to a Kerberos service using the user's
+principal name.
+
+.. note::
+   The method refers to the ``GSSAPI`` authentication mechanism instead
+   of ``Kerberos`` because the driver authenticates using the
+   `GSSAPI RFC-4652 <https://tools.ietf.org/html/rfc4752>`_  SASL
+   mechanism.
+
+The following code snippets show how to specify the authentication mechanism,
+using the following placeholders:
+
+* ``username`` - your URL-encoded principal name, e.g. "username%40REALM.ME"
+* ``hostname`` - network address of your MongoDB server, accessible by your client
+* ``port`` - port number of your MongoDB server
+
+Select the :guilabel:`Connection String` or the :guilabel:`MongoCredential`
+tab below for instructions and sample code for specifying this authentication
+mechanism:
+
+.. tabs::
+
+   .. tab::
+      :tabid: Connection String
+
+      To specify the GSSAPI authentication mechanism using a connection
+      string, assign the ``authMechanism`` parameter to the value
+      ``GSSAPI`` in your connection string. Set the ``authSource`` parameter
+      to the value ``$external``. Your code to instantiate a ``MongoClient`` 
+      should resemble the following:
+
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create("<username>@<hostname>:<port>/?authSource=<authenticationDb>&authMechanism=GSSAPI");
+
+   .. tab::
+      :tabid: MongoCredential
+
+      To specify the GSSAPI authentication mechanism using the
+      ``MongoCredential`` class, use the ``createGSSAPICredential()``
+      method. Your code to instantiate a ``MongoClient`` should resemble the
+      following:
+
+      .. include:: /includes/fundamentals/code-snippets/auth-credentials-gssapi.rst
+
+
+In order to acquire a Kerberos ticket, the GSSAPI Java libraries may require
+that you specify the realm and Key Distribution Center (KDC) properties. See
+the sample settings in the example below:
+
+.. code-block:: properties
+
+   java.security.krb5.realm=MYREALM.ME
+   java.security.krb5.kdc=mykdc.myrealm.me
+
+You may need to specify one or more of the following additional properties
+depending on your Kerberos setup:
+
+- ``SERVICE_NAME``
+- ``CANONICALIZE_HOST_NAME``
+- ``JAVA_SUBJECT``
+- ``JAVA_SASL_CLIENT_PROPERTIES``
+
+Select the :guilabel:`Connection String` or the :guilabel:`MongoCredential`
+tab below for instructions on how to specify these properties:
+
+.. tabs::
```

Comment:
This was a little tough to organize as these settings are independent of which method you use to build the connection and I didn't want to repeat them. The information in the tabs is specific to the method used to build the connection, so that should be separated into the tabs. Does that make sense, or did I misread your comment?

Reply:
I think there should only be a single tab set containing one tab for Connection String and one for MongoCredential. We have two sets of these tabs right now for this auth mechanism. You can accomplish this by deleting the second tab set and moving information relevant to each method to it's previously defined tab set, or making the second tab set hidden and adjusting the text in the paragraphs to compensate.

Reply:
Oh, I think I see what you mean. I didn't know that redeclaring `.. tabs::` created a new independent set of tabs.